encoding/bool_encode.go

```diff
@@ -1,24 +1,42 @@
 package encoding
 
 import (
-	"errors"
 	"io"
 )
 
 // BoolEncoder encodes bool values.
 type BoolEncoder interface {
 	// Encode encodes a collection of bools and writes the encoded bytes to the writer.
-	// Callers should explicitly call `Reset` before subsequent call to `Encode`.
-	Encode(writer io.Writer, values ForwardBoolIterator) error
+	Encode(writer io.Writer, valuesIt ForwardBoolIterator) error
 }
 
 // BoolEnc is a bool encoder.
-type BoolEnc struct{}
+type BoolEnc struct {
+	buf []byte
+	t   []byte
+	f   []byte
+}
 
 // NewBoolEncoder creates a new bool encoder.
-func NewBoolEncoder(writer io.Writer) *BoolEnc { return &BoolEnc{} }
+func NewBoolEncoder() *BoolEnc {
+	return &BoolEnc{
+		t: []byte{1},
+		f: []byte{0},
+	}
+}
 
 // Encode encodes a collection of bools and writes the encoded bytes to the writer.
-func (enc *BoolEnc) Encode(writer io.Writer, values ForwardBoolIterator) error {
-	return errors.New("not implemented")
+func (enc *BoolEnc) Encode(writer io.Writer, valuesIt ForwardBoolIterator) error {
+	return runLengthEncodeBool(writer, &enc.buf, enc.marshalFn, valuesIt)
```

Comment:
Do we want to support bit packing for booleans? Feels it could help greatly to reduce space for booleans where RLE doesn't necessarily apply.